Gems/LmbrCentral/Code/Source/Shape/PolygonPrismShape.h

```diff
@@ -103,12 +104,78 @@ namespace LmbrCentral
             AZStd::vector<AZ::Vector3> m_triangles; ///< Triangles comprising the polygon prism shape (for intersection testing).
         };
 
+        // PolygonPrism has a complicated set of access patterns, so it needs specialized mutex logic to be able to use a shared_mutex
+        // with this shape type. The shared_mutex is desirable because it allows the shape to be queried from multiple threads at once,
+        // while still protecting reads from running concurrently with writes.
+        // The complication is that the PolygonPrismShape has an underlying PolygonPrism, which is the actual data container.
+        // PolygonPrism itself has optional callbacks that trigger when data is modified. The EditorPolygonPrismShapeComponent uses those
+        // callbacks to query the PolygonPrismShape to refresh some data.
+        // This means that when modifying data, the PolygonPrismShape can lock a unique_lock, but while it's holding that lock, callbacks
+        // can trigger that try to read data on the same thread, which will try to lock a shared_lock.
+        // The extra logic the PolygonPrism*LockGuard classes perform is that they track the thread that acquires a unique lock,
+        // and they don't try to acquire or release a shared lock if one is requested on the same thread as the unique lock.
+
+        class PolygonPrismSharedLockGuard
+        {
+        public:
+            explicit PolygonPrismSharedLockGuard(AZStd::shared_mutex& mutex, AZStd::thread::id& uniqueLockThreadId)
```

Comment:
explicit is not need for a function that accepts two required parameters.

Reply:
Fixed!